mautic/mautic
# Fix emails in time filters

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

The emails in time widget worked fine until it was saved. When saved then filters with empty values were applied and the widget basically did not display anything.

Also the listener was expecting the filters in the `filter` array as that's how the API endpoint sends them but they are in the root of the `$params` array when the widget edit form sends them.

This PR introduces also simple `ArrayHelper` which should make he code smaller and easier to read.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Go to the dashboard.
2. Notice the emails in time widget shows some data. (if not, send some emails to get some testing data)
3. Edit the widget.
4. Change nothing and save.
5. The widget doesn't display any data.

#### Steps to test this PR:
1. Refresh the dashboard. Data are back.
2. Make sure that the filters work from the widget edit form as well as [from the API](https://developer.mautic.org/#%22emails-in-time%22-widget).
